exasol_transformers_extension/deployment/language_container_deployer_cli.py

```diff
@@ -30,33 +31,46 @@ def language_container_deployer_main(
         bucket: str,
         path_in_bucket: str,
         container_file: str,
+        version: str,
         dsn: str,
         db_user: str,
         db_pass: str,
         language_alias: str):
 
+    def call_runner():
+        LanguageContainerDeployer.run(
+            bucketfs_name=bucketfs_name,
+            bucketfs_host=bucketfs_host,
+            bucketfs_port=bucketfs_port,
+            bucketfs_use_https=bucketfs_use_https,
+            bucketfs_user=bucketfs_user,
+            bucketfs_password=bucketfs_password,
+            bucket=bucket,
+            path_in_bucket=path_in_bucket,
+            container_file=Path(container_file),
+            dsn=dsn,
+            db_user=db_user,
+            db_password=db_password,
+            language_alias=language_alias
+        )
+
     bucketfs_password = utils.get_password(
         bucketfs_password, bucketfs_user,
         utils.BUCKETFS_PASSWORD_ENVIRONMENT_VARIABLE, "BucketFS Password")
     db_password = utils.get_password(
         db_pass, db_user,
         utils.DB_PASSWORD_ENVIRONMENT_VARIABLE, "DB Password")
 
-    LanguageContainerDeployer.run(
-        bucketfs_name=bucketfs_name,
-        bucketfs_host=bucketfs_host,
-        bucketfs_port=bucketfs_port,
-        bucketfs_use_https=bucketfs_use_https,
-        bucketfs_user=bucketfs_user,
-        bucketfs_password=bucketfs_password,
-        bucket=bucket,
-        path_in_bucket=path_in_bucket,
-        container_file=Path(container_file),
-        dsn=dsn,
-        db_user=db_user,
-        db_password=db_password,
-        language_alias=language_alias
-    )
+    if container_file:
+        call_runner()
+    elif version:
+        with utils.get_container_file_from_github_release(version) as container:
+            container_file = container
+            call_runner()
+    else:
+        raise Exception("You should specify either the release version to "
+                        "download container file or the path of the already "
+                        "downloaded container file.")
```

Comment:
From readability point of view, the first version seems better for me. Why did you suggest this?

Reply:
Because a reader can think about the conditions one at the time and therefore from my point of view it is easier to understand when and what happens. Also it is clear when `call_runner()` is invoked. Also it is ensured that when `call_runner()` is invoked `container_file == True`, this is not guaranteed in the `elif` branch.

TL;DR: I think the individual parts are can be better reasoned about and we do not depend on an other function to ensure we don't crash.